src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int RESIZE_CONSTANT = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] internalArray;
+
+    public MyHashMap() {
+        internalArray = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
+        growIfArrayFull();
+        int hashedKey = hashKey(key);
+        int bucketNumber = hashedKey % internalArray.length;
```

Comment:
Probably you can combine this into one method hashedKey

Reply:
I think i can't, because in our node we have hash field. And if i will make that method, i can't add it. But if I will try  I need to implement hashCode and equals methods in node, and calculate hash 2 times with 1 hash function.